### ISSUE TITLE
test_stdio.StandardInputOutputTestCase.test_lastWriteReceived fails if a warning is printed during the tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2858]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2858|
|Type|defect|
|Created|2007-10-30 14:34:02Z|

This is a bad side effect.

Attachments:

* [2858.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aec/aec0f8f62c9a2434e6aaea6d069047b6de20ca73/1e5aa41fe0f9ce933ed9c4e5d7a9104c6dd515fa.diff) (569 bytes) - added by therve on 2007-10-30 14:47:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2858 2858
type__defect defect
reporter__therve therve
priority__highest highest
milestone__Python_2_6 Python-2.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1193754842000000 1193754842000000
changetime__1194115670000000 1194115670000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve removed owner|
|-|-|

Just to explain the problem: PTY fusions stdout and stderr, and warnings print to stderr, so the test fails. The attached fix corrects that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Counter-proposal in safer-lastwrite-test-2858 - just tries to be a bit safer by writing more bytes, in case a CPython error message ever decides to end with "x" one day.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

That's cool, it does the trick for me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun|
|-|-|

< therve> exarkun: I'm +1

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [21542]) Merge safer-lastwrite-test-2858

Author: exarkun
Reviewer: therve
Fixes [#2858](https://github.com/twisted/twisted/issues/2858)

Change the test method covering the reliability of the very last write on a pty
stdout so that it does not fail if the runtime generates warning messages on
stderr which get mixed up with the test program's output.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [21545]) Revert r21542 - test suite regression and obvious test defect

Most builders failed with this error:
```
===============================================================================
[FAIL]: twisted.test.test_stdio.StandardInputOutputTestCase.test_lastWriteReceived

Traceback (most recent call last):
  File "/srv/bb-slave/Run/slave/reactors/Twisted/twisted/test/test_stdio.py", line 106, in eb
    return callback(err)
  File "/srv/bb-slave/Run/slave/reactors/Twisted/twisted/test/test_stdio.py", line 156, in processEnded
    "Received %r from child, did not find expected bytes.")
twisted.trial.unittest.FailTest: Received %r from child, did not find expected bytes.
-------------------------------------------------------------------------------
```
refs [#2858](https://github.com/twisted/twisted/issues/2858)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

This is probably fixed now.  I think it was basically a Combinator bug/misfeature which caused the child process the test starts to use the Combinator-managed version of Twisted instead of the version of Twisted buildbot checked out.  I worked around this by passing the magic string on the command line instead of importing it.  I also fixed a bug where the reactor argument was ignored by the particular script modified here.

Still in **safer-lastwrite-test-2858**

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

Ok that was a weird problem. The buildbot is green, so please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [21558]) Merge safer-lastwrite-test-2858

Author: exarkun
Reviewer: therve
Fixes [#2858](https://github.com/twisted/twisted/issues/2858)

Change the test method covering the reliability of the
very last write on a pty stdout so that it does not fail
if the runtime generates warning messages on stderr which
get mixed up with the test program's output.

Also change stdio_test_lastwrite.py so that it uses the
reactor specified on its command line instead of always
using the default reactor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
